File: thirdWeek/BinaryTreeInorderTraversal.java
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> valueList = new LinkedList<>();
        Stack<TreeNode> stack = new Stack<>();
        while (!stack.isEmpty() || root != null) {
            if (root != null) {
                stack.push(root);
                root = root.left;
            } else {
                root = stack.pop();
                valueList.add(root.val);
                root = root.right;
            }
        }
        
        return valueList;
    }
}

File: thirtyThirdWeek/binary_tree_inorder_traversal.py
class Solution:
    def inorderTraversal(self, root: TreeNode) -> List[int]:
        node = root
        stack = []
        ans = []
        while stack or node:
            if node:
                stack.append(node)
                node = node.left
            else:
                node = stack.pop()
                ans.append(node.val)
                node = node.right
        return ans
